src/main/java/com/coveo/pushapiclient/CatalogSource.java

```diff
@@ -0,0 +1,17 @@
+package com.coveo.pushapiclient;
+
+public class CatalogSource {
```

Comment:
Catalog feels like a commerce term. Should we use StreamSource instead? Interesting to see that the [url](https://docs.coveo.com/en/3295/index-content/add-or-edit-a-catalog-source#stream-api-url) mentions `stream` and does not include `catalog`.

Reply:
Agreed. Can we still use that endpoint to upload to a different source than the catalog? Either way, let's stay consistent with the API and rename the class.
Thanks!

Reply:
Yea, I believe the stream api is generic/reusable. Today I think it's only used by the catalog source.
@hdhayneCoveo or @jmorissette, do you know if there's another source using it?

One downside of the generic name might be that it's less clear that catalog source -> stream api, but maybe we can alleviate this with a docs code sample. It feels more correct for the sdk to use the generic name 🤞

Reply:
nop i am not aware of any other source than the catalog source

Reply:
After going over the other threads, I am still not satisfied with the class naming.

If we decide to take `StreamSource` rather than `CatalogSource`, then the upload services classes (FullUploadStream, UpdateStream) do not make a lot of sense:

Will it be clearer if we stick with `CatalogSource` and rename services to:
1. `PushDocumentServiceManager`
2. `StreamDocumentServiceManager`
Or something like that to differentiate the 2 APIs (I am just following the naming conventions used in the [.NET SDK](https://github.com/coveooss/platform-sdk)).

This could make sense is we later support in the SDK another source that is also streamable:
```java
CatalogSource source= new CatalogSource("uri", "token");
PushDocumentServiceManager push = new PushDocumentServiceManager(source);
StreamDocumentServiceManager stream = new StreamDocumentServiceManager(source);

AnotherSource sourceB= new AnotherSource("uri", "token");
StreamDocumentServiceManager streamB = new StreamDocumentServiceManager(sourceB);
```

Reply:
`CatalogSource` would be clearer too for developers creating the source in Admin. I like your idea ✔️ 
Could the names of the manager classes be shortened?
e.g. `new PushService(source)`, `new StreamService(source)`

Reply:
Perfect! I think we got it :D